docs/docs/guides/multi-stage-delivery.md

```diff
@@ -0,0 +1,258 @@
+
+NAME                           APPNAME        VERSION   PHASE
+podtato-head-v0.3.0-6b86b273   podtato-head   v0.3.0    Completed
+```
+
+And there will be another `KeptnAppVersion` in the `production` stage.
+
+```shell
+$ kubectl get keptnappversion -n podtato-head-production
+
+NAME                           APPNAME        VERSION   PHASE
+podtato-head-v0.3.0-6b86b273   podtato-head   v0.3.0    Completed
+```
+
+## Promoting a workload
+
+Now that everything is set up, it is time to deploy a new version into `dev`
+and automatically promote it into `production` if the deployment into `dev`
+has been successful.
+To do so, change the `serviceVersion` value in `dev/values.yaml` from `v0.3.0`
+to `v0.3.1`:
+
+```yaml title="dev/values.yaml"
+{% include "./assets/multi-stage-delivery/values-dev-updated.yaml" %}
+```
+
+After changing the value, commit and push the updated file to
+the upstream repository:
+
+```shell
+git add dev/values.yaml
+git commit -m "deploy new version into dev"
+git push
+```
+
+Eventually, Argo CD will notify these changes and apply the updated helm chart
+to the cluster.
+This is also reflected in a new `KeptnAppVersion` being created in the `dev` stage:
+
+```shell
+$ kubectl get keptnappversion -n podtato-head-dev
+
+NAME                           APPNAME        VERSION   PHASE
+podtato-head-v0.3.0-6b86b273   podtato-head   v0.3.0    Completed
+podtato-head-v0.3.1-d4735e3a   podtato-head   v0.3.1    Completed
+```
+
+After the new version has been deployed, and the post deployment phase
+is completed, the `KeptnTask` executed in the promotion phase
+triggers a GitHub action that creates a pull request for applying the
+updated values to the helm chart of the `production` stage.
+You will see the PR in the *Pull requests* section of your upstream repository:
+
+![PR for promoting the new version](./assets/multi-stage-delivery/dev2prod-pr.png)
+
+After approving and merging the pull request into `main`,
+Argo CD eventually synchronizes the application in the
+`production` stage, and the updated version is deployed.
+This is reflected in a new `KeptnAppVersion` being created
+in the `podtato-head-production` namespace:
+
+```shell
+$ kubectl get keptnappversion -n podtato-head-production
+
+NAME                           APPNAME        VERSION   PHASE
+podtato-head-v0.3.0-6b86b273   podtato-head   v0.3.0    Completed
+podtato-head-v0.3.1-d4735e3a   podtato-head   v0.3.1    Completed
+```
+
+## Inspecting the Deployment Traces
+
+To keep track of how a workload version progresses through
+the stages, we passed through the OTel span id of the
+promotion phase in the `dev` stage to the applied
+`KeptnAppContext` in the `production` stage.
+Due to that, the deployment trace of the promoted version
+in the `production` stage contains the reference to the
+promotion phase, which is also visible in the trace visualization in Jaeger:
+
+![Deployment Trace](./assets/multi-stage-delivery/trace.png)
```

Comment:
I think a conclusion section would be good here, that repeats what the users just went through.